src/coreclr/System.Private.CoreLib/src/System/Runtime/CompilerServices/CastHelpers.cs

```diff
@@ -214,24 +214,26 @@ private static CastResult TryGet(nuint source, nuint target)
                 if (interfaceCount != 0)
                 {
                     MethodTable** interfaceMap = mt->InterfaceMap;
-                    for (nuint i = 0; ; i += 4)
+                    nuint i = 0;
+                    for (; i < interfaceCount - 4; i += 4)
                     {
-                        if (interfaceMap[i + 0] == toTypeHnd)
+                        MethodTable** map = interfaceMap + 4;
+                        if (interfaceMap[0] == toTypeHnd ||
+                            interfaceMap[1] == toTypeHnd ||
+                            interfaceMap[2] == toTypeHnd ||
+                            interfaceMap[3] == toTypeHnd)
```

Comment:
wonder if we can use Simd here, hover it might regress the cases where the interface of interest is the first one (or second)

Reply:
> Also regarding `nterfaceCount - 4` check - will this block be executed if `interfaceCount == 4`?

No, changed to gteq

Reply:
> wonder if we can use Simd here, however, it might regress the cases where the interface of interest is the first one (or second)

Comparing to vector is fast; but then need to use `PMOVMSKB` or `TESTZ` to extract a usable condition for the `if` which throws in a 3 clock latency... 🤔

Reply:
Difference in loop body would be between these two for simd?
```asm
G_M4132_IG04:
       lea      r9, [rdx+32]
       cmp      qword ptr [rdx], rsi
       je       SHORT G_M4132_IG08
       cmp      qword ptr [rdx+8], rsi
       je       SHORT G_M4132_IG08
       cmp      qword ptr [rdx+16], rsi
       je       SHORT G_M4132_IG08
       cmp      qword ptr [rdx+24], rsi
       je       SHORT G_M4132_IG08
       mov      rdx, r9
       add      r8, 4
       cmp      r8, rcx
       jle      SHORT G_M4132_IG04
```
```asm
G_M4132_IG04:
       lea      r9, [rdx+32]
       mov      rdx, qword ptr [rdx]
       vpcmpeqq ymm1, ymm0, ymmword ptr[rdx]
       vpmovmskb edx, ymm1
       test     edx, edx
       je       SHORT G_M4132_IG08
       mov      rdx, r9
       add      r8, 4
       cmp      r8, rcx
       jle      SHORT G_M4132_IG04
```